src/json_parser.h

```diff
@@ -25,7 +25,10 @@ class JSONParser {
   // We might want a lighter-weight JSON parser for this use case. But for now
   // using V8 is good enough.
   RAIIIsolate isolate_;
+
+  v8::Isolate::Scope isolate_scope_;
```

Comment:
I feel the Isolate::Scope should be created inside the Parse(), GetTopLevelStringField() and GetTopLevelBoolField() methods.

That JSONParser uses an isolate internally is an implementation detail. Creating a JSONParser instance should not change the program's ambient state (the entered isolate.)

Reply:
I'm inclined to agree, but note that JSONParser currently changes the handle scope and the context scope.

Reply:
Oh, so it does. Okay, better tackled in another PR then.